hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
> So the key difference between HTTP/1.1 and HTTP/1.0 responses for that matter, as I see it, is that HTTP/1.1 response must include a connection: close header if a server is about to close connection in the absence of content-length/chunked encoding headers

This is true in the sense that the spec requires it, but the spec also requires that in the absence of all three headers (no `Connection: close`, no `Transfer-Encoding: chunked`, and no `Content-Length`), the response must be treated *as if* `Connection: close` were present. So my suggestion is that we take that route, rather than special-case HTTP/1.0.